### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by JohnFTitor**
# Hi there 👋🏻 

Thanks for taking the time to review the project. Here are some things important to note about it:

### Changes implemented in "Set Up Associations" branch 

- ✅  Create a has-many/belongs-to relationship between Classroom and Students.
- ✅  Implemented update functionality between Classroom and Students (One modifies the other).
- ✅  Create a many-to-many relationship between Books and Person, using the intermediary class Rental 

## Final notes 📔 

Is expected that this project meets the requirements. Please let me know if anything is left to be done or can be improved. Again, thanks for the review.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rents
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rents = []
+  end
+end
```
Comment (Gambit142) [inline book.rb]: - [ ] In your `book.rb`, kindly add a method that creates a new instance of the `Rental` class. Remember the relationship between `Book` and `Person` is a many-many relationship and they are joined through `Rentals`. This means that for a `Book` to connect with the `Person`, a `Book` should have many `Rentals` and a `Person` should have many `Rentals`. Kindly create this method that creates a new instance of Rental in the `Book` class.

#### _**Hint:**_ The method should have parameters for an instance of `Person`, and the `date`. Then when you create the instance of `Rental`, you can pass `self`. 👍

File: person.rb [hunk 2]
```diff
@@ -2,14 +2,15 @@
...
+  attr_reader :id, :rents
 
   def initialize(age, parent_permission: true, name: 'Unknown')
     super()
     @id = Random.rand(0..1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rents = []
   end
```
Comment (Gambit142) [inline person.rb:16]: - [ ] In your `person.rb`, kindly add a method that creates a new instance of the `Rental` class. Remember the relationship between `Book` and `Person` is a many-many relationship and they are joined through `Rentals`. This means that for a `Book` to connect with the `Person`, a `Book` should have many `Rentals` and a `Person` should have many `Rentals`. Kindly create this method that creates a new instance of Rental in the `Person` class. 

#### _**Hint:**_ The method should have parameters for an instance of `Book`, and the `date`. Then when you create the instance of `Rental`, you can pass `self`. 👍

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @JohnFTitor 👋, 

### Highlights
- Implemented the belongs-to and has-many relationship properly 💯 
- Created classes properly 💯 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: book.rb (hunk 1 above)
Comment (JohnFTitor) [reply book.rb]: Roger!

File: person.rb (hunk 2 above)
Comment (JohnFTitor) [reply person.rb:16]: Roger!

**Comment by omar-labana** [comment]
# **Approved ✔️**

Hi @JohnFTitor,

Congratulations on completing this project! 👏🏼 🎉

You have correctly implemented all of the requirements📃 correctly and without any errors⚠️ or bugs🪲.

Cheers and good luck in your next milestones! 🍀

Feel free to leave any questions or comments in the PR thread(don't forget to tag me 😄) if something is not 100% clear.

---
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**